hawkbit-repository/hawkbit-repository-jpa/src/main/java/org/eclipse/hawkbit/repository/jpa/ActionRepository.java

```diff
@@ -64,6 +64,27 @@
      */
     Page<Action> findByDistributionSetId(Pageable pageable, Long dsId);
 
+    /**
+     * Retrieves all active {@link Action}s which are referring the given
+     * {@link DistributionSet}.
+     *
+     * @param set
+     *            the {@link DistributionSet} on which will be filtered
+     * @return the found {@link Action}s
+     */
+    List<Action> findByDistributionSetAndActiveIsTrue(DistributionSet set);
+
+    /**
+     * Retrieves all active {@link Action}s which are referring the given
+     * {@link DistributionSet} and are not in CANCELING state
+     *
+     * @param set
+     *            the {@link DistributionSet} on which will be filtered
+     * @return the found {@link Action}s
+     */
+    @Query("Select a from JpaAction a where a.distributionSet = :set and a.active = true and not a.status = org.eclipse.hawkbit.repository.model.Action.Status.CANCELING ")
+    List<Action> findByDistributionSetAndActiveIsTrueAndActionStatusIsNotCanceling(DistributionSet set);
```

Comment:
please specify the Action Status as method param and leave it to the management class to decide on the particular value (in this case CANCELING)